libclc/opencl/lib/generic/atomic/atom_add.cl

```diff
@@ -6,32 +6,35 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <clc/atomic/clc_atomic_fetch_add.h>
 #include <clc/opencl/atomic/atom_add.h>
-#include <clc/opencl/atomic/atomic_add.h>
+
+// Non-volatile overloads are for backward compatibility with OpenCL 1.0.
```

Comment:
I don't remember the history here, but isn't this backwards? Even if the language is volatile qualified we probably shouldn't be emitting IR with volatile markers